pkg/pipeline/templates/controller.go.tmpl

```diff
@@ -37,9 +43,13 @@ func Setup(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter, s terra
 		managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))),
 		managed.WithFinalizer(terraform.NewWorkspaceFinalizer(ws, xpresource.NewAPIFinalizer(mgr.GetClient(), managed.FinalizerName))),
 		managed.WithTimeout(3*time.Minute),
+		{{- if .Initializers }}
+		managed.WithInitializers(initializers),
+		{{else}}
 		{{- if .DisableNameInitializer }}
 		managed.WithInitializers(),
 		{{- end}}
+		{{- end}}
```

Comment:
@muvaf We may discuss the possible issues that can occur the new logic about adding the WithInitializers option.